### DISCUSSION TITLE
Access control for cached images?

### DISCUSSION THREAD

**Discussion by gabor-kovacs** [vercel/next.js#60308] [Help]
I have non-public images in an S3 bucket. I get a signed url valid for X time and pass that to the next `<Image>` component.

The issue with this is that
- The images are cached in the _next folder and these are publically available given the url
- The signed url query parameter changes on every page load and the images are treated different and cached again under a different folder

Is there a way to block public access to the cached images?
I am already using next-auth.

Is there any alternative to serve optimized images from an S3 bucket and caching them in the browser but not on the nextjs server?

**Comment by don-esteban** [comment]
Did you read this? https://nextjs.org/docs/pages/api-reference/components/image

Perhaps you can control the caching appropriately? Or delete cache manually? Or resort to a custom loader?

**Reply by gabor-kovacs** [reply]
I have read the docs, it describes the default loader caching behavior but there is no way to control it.
I tried setting cache-control headers of the upstream images and setting minimalTTL to 0 but its ignored by next.

Deleting the cache manually might be possible but its not ideal and would create a lot of i/o operations.

I tried using a custom loader which solves the caching issue, but the Image component only accepts strings. I tired to use createObjectUri on a blob and pass it which didn't work and passing it as a base64 image which did work but not what i want.

I would like to use the benefits of image transformations without caching the images on the next server.

[This](https://github.com/vercel/next.js/discussions/25846) has been discussed before but the suggested method is with the old /pages api.
